### PR TITLE
Applied admin-only methods so that normal users are restricted.

### PR DESCRIPTION
**Pull request by lecegues**
### Description
- Used the pre-made functions (authenticate.js, userContext) to restrict the user from doing admin-only actions if they don't have the proper credentials
- Backend: `authenticate.js` middleware to prevent requests from browsers without a valid token
- Frontend: `userContext.js` to hide admin-only buttons (e.g. add, delete, modify) from normal users

### CODE REVIEW

**Comment by lecegues** [comment]
### Note
An integration test was removed due to the need of a login test-hook ( commit 2116823eed2f68d36a10eb242e084b977ef4e647).
This test will be added back in a patch for issue #123

File: backend/routes/artifactTypes.js
Comment (funmise) [inline backend/routes/artifactTypes.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/artifacts.js
Comment (funmise) [inline backend/routes/artifacts.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/baseShapes.js
Comment (funmise) [inline backend/routes/baseShapes.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/bladeShapes.js
Comment (funmise) [inline backend/routes/bladeShapes.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/catalogues.js
Comment (funmise) [inline backend/routes/catalogues.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/crossSections.js
Comment (funmise) [inline backend/routes/crossSections.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/cultures.js
Comment (funmise) [inline backend/routes/cultures.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/haftingShapes.js
Comment (funmise) [inline backend/routes/haftingShapes.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/materials.js
Comment (funmise) [inline backend/routes/materials.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/periods.js
Comment (funmise) [inline backend/routes/periods.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/projectilePoints.js
Comment (funmise) [inline backend/routes/projectilePoints.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/regions.js
Comment (funmise) [inline backend/routes/regions.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/sites.js
Comment (funmise) [inline backend/routes/sites.js:1]: need to update docs to include authenticateAdmin param

File: backend/routes/users.js
Comment (funmise) [inline backend/routes/users.js:1]: need to update docs to include authenticateAdmin param

**Review by funmise — APPROVED** [review]
Looks good. Should probably update all the docs with the new param

**Review by brj496 — APPROVED** [review]
Looks good, I don't see any particular issues with the code.

**Pull request merged**